src/Microsoft.AspNetCore.Mvc.RazorPages/Infrastructure/PageDirectiveFeature.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+
+namespace Microsoft.AspNetCore.Mvc.RazorPages.Infrastructure
+{
+    public static class PageDirectiveFeature
+    {
+        public static bool TryGetRouteTemplate(RazorProjectItem projectItem, out string template)
+        {
+            const string PageDirective = "@page";
+
+            string content;
+            using (var streamReader = new StreamReader(projectItem.Read()))
+            {
+                content = streamReader.ReadToEnd();
+            }
+
+            if (content.StartsWith(PageDirective, StringComparison.Ordinal))
```

Comment:
The answer is most likely no, given that you are the expert you definitely known better. This is in essence a mini/super incomplete parser for razor, would it be possible to use razor's parser for this taks? (Or lexer if it has one). I'm concerned for changes in behavior in this area would need to be replicated here, I also can see users putting `@page` after things like @using or @model, and as with any code involving parsing, all the small details about it (whitespace handling, etc) that are already baked in the parser and we should reuse if possible